src/OTSubscriber.js

```diff
@@ -81,18 +81,22 @@ export default class OTSubscriber extends Component {
   }
   render() {
     const containerStyle = this.props.containerStyle;
-    const childrenWithStreams = this.state.streams.map((streamId) => {
-      const streamProperties = this.props.streamProperties[streamId];
-      const style = isEmpty(streamProperties) ? this.props.style : (isUndefined(streamProperties.style) || isNull(streamProperties.style)) ? this.props.style : streamProperties.style;
-      return <OTSubscriberView key={streamId} streamId={streamId} style={style} />
-    });
-    return <View style={containerStyle}>{ childrenWithStreams }</View>;
+    if (!this.props.children) {
+      const childrenWithStreams = this.state.streams.map((streamId) => {
+        const streamProperties = this.props.streamProperties[streamId];
+        const style = isEmpty(streamProperties) ? this.props.style : (isUndefined(streamProperties.style) || isNull(streamProperties.style)) ? this.props.style : streamProperties.style;
+        return <OTSubscriberView key={streamId} streamId={streamId} style={style} />
+      });
+      return <View style={containerStyle}>{ childrenWithStreams }</View>;
+    }
+    return this.props.children(this.state.streams);
```

Comment:
I think it should be `return this.props.children(this.state.streams) || null;`
Regarding your example, in your renderSubscribers function there is a case when there's no streams the state, so your function won't return anything and the app will crush. In react you need at least to return a null value.

Reply:
What if we put the `.map` within the render function of`OTSubscriber` component? So all the validations are handled by the library.

I was thinking something like this:

```javascript
// OTSubscriber.js

render() {
    const containerStyle = this.props.containerStyle;
    console.log('this.props.streamProperties', this.props)
    if (!this.props.children) {
        const childrenWithStreams = this.state.streams.map((streamId) => {
          const streamProperties = this.props.streamProperties[streamId];
          const style = isEmpty(streamProperties) ? this.props.style : (isUndefined(streamProperties.style) || isNull(streamProperties.style)) ? this.props.style : streamProperties.style;
          return <OTSubscriberView key={streamId} streamId={streamId} style={style} />
        });
        return <View style={containerStyle}>{ childrenWithStreams }</View>;
    }
    const child = this.props.children;
    const childrenWrapper = Array.isArray(this.state.streams)
    ? this.state.streams.map(stream => (child && typeof child === 'function' ? child(stream) : child))
    : null;
    return childrenWrapper;
}

```
And then an example of usage would be: 

```javascript 

// App.js

<OTSubscriber
        streamProperties={streamProperties}>
        {this.renderSubscribers}
      </OTSubscriber>

      ...

      renderSubscribers = (streamId) => {
            <TouchableOpacity
              onPress={() => this.handleStreamPress(streamId)}
              key={streamId}
              style={subscriberStyle}
            >
              <OTSubscriberView streamId={streamId} style={subscriberStyle} />
            </TouchableOpacity>
      }
```

Reply:
I agree with @ggoldens in that we should return a null value, but I don't think additional validation is necessary. PropTypes will indicate to the user that the children prop should be a function. We could check that `this.state.streams` is an array, though. As for whether we should use`.map` within ´OTSubscriber´or just call the function and pass in the array, I don't really have a preference.

Reply:
I definitely think that the map should be done outside the lib in the render closure to give you more control over the rendering. E.g. maybe you want to style every other.

Reply:
That's precisely the case we have: we might want to style the view based on the participant audio level for instance.